src/test/java/teammates/test/cases/webapi/BinCourseActionTest.java

```diff
@@ -0,0 +1,116 @@
+package teammates.test.cases.webapi;
+
+import java.util.List;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.CourseAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.util.Const;
+import teammates.logic.core.CoursesLogic;
+import teammates.ui.webapi.action.BinCourseAction;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.output.MessageOutput;
+
+/**
+ * SUT: {@link BinCourseAction}.
+ */
+public class BinCourseActionTest extends BaseActionTest<BinCourseAction> {
+
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.BIN_COURSE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return PUT;
+    }
+
+    @Override
+    @Test
+    protected void testExecute() throws Exception {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        String instructorId = instructor1OfCourse1.googleId;
+
+        loginAsInstructor(instructorId);
+
+        ______TS("Not enough parameters");
+
+        verifyHttpParameterFailure();
+
+        ______TS("Typical case, 2 courses");
+
+        logic.createCourseAndInstructor(instructorId, "icdct.tpa.id1", "New course", "UTC");
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+
+        BinCourseAction deleteAction = getAction(submissionParams);
+        JsonResult result = getJsonResult(deleteAction);
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        MessageOutput msg = (MessageOutput) result.getOutput();
+        assertEquals("The course idOfTypicalCourse1 has been deleted. You can restore it from the Recycle Bin manually.",
+                msg.getMessage());
+
+        List<CourseAttributes> courseList = logic.getCoursesForInstructor(instructorId);
+        assertEquals(1, courseList.size());
+        assertEquals("icdct.tpa.id1", courseList.get(0).getId());
+
+        ______TS("Masquerade mode, delete last course");
+
+        loginAsAdmin();
+
+        submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, "icdct.tpa.id1",
+        };
+
+        deleteAction = getAction(addUserIdToParams(instructorId, submissionParams));
+        result = getJsonResult(deleteAction);
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        msg = (MessageOutput) result.getOutput();
+        assertEquals("The course icdct.tpa.id1 has been deleted. You can restore it from the Recycle Bin manually.",
+                msg.getMessage());
+
+        courseList = logic.getCoursesForInstructor(instructorId);
```

Comment:
The verification is implicit. Why not using `logic.getCourse` to verify the status directly?